alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultUserAccessor.java

```diff
@@ -118,14 +118,18 @@ public UserModel updateUser(UserModel user, boolean passwordEncoded) throws Aler
         UserEntity existingUser = userRepository.findById(userId)
                                       .orElseThrow(() -> new AlertDatabaseConstraintException(String.format("No user found with id '%s'", userId)));
         Long existingUserId = existingUser.getId();
-
-        String password = passwordEncoded ? user.getPassword() : defaultPasswordEncoder.encode(user.getPassword());
-        UserEntity newEntity = new UserEntity(user.getName(), password, user.getEmailAddress(), user.isExpired(), user.isLocked(), user.isPasswordExpired(), user.isEnabled(), existingUser.isExternal());
-        newEntity.setId(existingUserId);
+        UserEntity savedEntity = existingUser;
+        // if it isn't an external user then update username, password, and email.
+        if (!existingUser.isExternal()) {
```

Comment:
Should we expose something to the user to let them know that the username, password, and email were not updated?